lightstep/resource_metric_condition.go

```diff
@@ -257,11 +257,19 @@ func getQuerySchema() map[string]*schema.Schema {
 			Optional: true,
 		},
 		"tql": {
-			Type:     schema.TypeString,
-			Optional: true,
-			Computed: true,
+			Deprecated:  "Use lightstep_dashboard or query_string field instead",
+			Description: "Deprecated, use query_string instead",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
 		},
 		"spans": getSpansQuerySchema(),
+		"query_string": {
+			Type:        schema.TypeString,
+			Description: "query expressed in Lightstep Unified Query Language (UQL)",
+			Optional:    true,
+			Computed:    true,
+		},
```

Comment:
I was on the fence about whether to add `query_string` to this resource or simply have the deprecation message say "use the lightstep_dashboard resource instead". I decided to provide both, but am glad to change that if there's a strong preference.

Reply:
the addition of this field and the deprecation of the `tql` field are the only changes needed to address the "it's-not-called-tql-anymore" problem, right? 

the creation of the `lightstep_dashboard` resource addresses a different problem (the use of `metrics` when we mean `unified`). i think the latter problem is also worth fixing, but it might've been good to split into a separate PR; this one is a lot to review

Reply:
what's the plan for releasing v2? i have some concern about the `lightstep_dashboard` resource, just because it leaves us in a hard-to-maintain duplicated state until we get around to removing `lightstep_metric_dashboard`. but if you're planning on doing that real soon then it's probably not much of a concern

it's nice that folks get to migrate to v2-compatible hcl before we actually release it, but in practice i dont think `s/lightstep_metric_dashboard/lightstep_dashboard` and `s/tql/query_string` represent huge migration burdens for folks looking to upgrade when the time comes. did you consider just adding `lightstep_dashboard` and removing `lightstep_metric_dashboard` as part of the v2 breaking changes?

Reply:
A bunch of thoughts...

(1) I don't think this creates a  "hard-to-maintain duplicated state" as we shouldn't be updating `lightstep_metric_dashboard` anymore. I suspect the forked file duplicated code is giving the impression this is more complex than it is (the "resources" are shallow interfaces on the same underlying `Client` code).  I'll consider removing the forking to see if I can make it more clear this is basically a renaming and TF fields/state clean-up.

(2) From backwards compatibility work in the past and do have a somewhat strong opinion that the "better" way to do binary breaks is to introduce the new interface first, have a period of overlap with both available (basically the deprecation period), then removing the old one. I don't think removing an old interface and introducing a new one at the same time makes for as nice a developer experience as it's harder to do incremental migrations or test configurations with partial migrations (e.g. move one dashboard, run it side-by-side, make sure it didn't change anything).  Also, it's a bit more than "just a string substitution" since the resource changes the user's underlying Terraform state.

(Aside: I didn't mark _all_ the old lightstep_metric_dashboard query fields as deprecated since I'm not sure exactly how Terraform treats deprecated fields, but it _seems_ like it might force the user to fix all those before `apply` can be run again -- and per my point about being able to migrate w/o having to do it wholesale, I wanted to avoid that.)

Reply:
👍 this all sounds good!